### DISCUSSION TITLE
Reading the source code...

### DISCUSSION THREAD

**Discussion by efecollie** [hneemann/Digital#1095] [Q&A]
Hi,
I want to understand the internal workings of the digital. Any advice on that? 
I think eventually I need to read the source code. Where to start, and which parts of the source code can be considered as "core"?
Thanks in advance,

**Comment by hneemann** [comment]
The README.md file that is on GitHub gives a very rough overview of the function and the differences to Logisim. There is a package "core" which contains all elementary components of the simulation. You can get a good feel for how the simulator is implemented by looking at the "Node" class. It is the parent class of all active components. If you want to understand how the circuit consisting of the individual components and the wires is turned into a model that can be simulated, you can look at the class "ModelCreator". However, unlike the class "Node", this is a very complex process, where some special cases have to be considered, and therefore not so simple. Each time the simulation is started, a new model is created.

If you have specific questions, you can always contact me.

**Comment by efecollie** [comment]
Dear Helmut, Thank you very much for your very kind comments..
